mmv1/templates/terraform/pre_update/storage_insights_dataset_config.go.tmpl

```diff
@@ -0,0 +1,116 @@
+updateMask := []string{}
+
+if d.HasChange("include_newly_created_buckets") {
+	updateMask = append(updateMask, "includeNewlyCreatedBuckets")
+}
+
+if d.HasChange("retention_period_days") {
+	updateMask = append(updateMask, "retentionPeriodDays")
+}
+
+if d.HasChange("description") {
+	updateMask = append(updateMask, "description")
+}
+
+if d.HasChange("include_cloud_storage_locations") {
+	_, new_storage_locations := d.GetChange("include_cloud_storage_locations")
+	if new_locations, ok := new_storage_locations.([]interface{}); ok && len(new_locations) > 0 {
+        updateMask = append(updateMask, "includeCloudStorageLocations")
```

Comment:
only nit-pick: can you fix the indentation in this file? Should be tabs instead of spaces in this file. thanks!